### PR TITLE
fix: add utility function to get mta path

### PR DESCRIPTION
**Pull request by slin-sap**
#1126

Add utility function to get mta.yaml config file path if the input Fiori app path belong to a MTA project.

### CODE REVIEW

File: packages/project-access/src/constants.ts [hunk 1]
```diff
@@ -8,5 +8,6 @@ export const FileName = {
     Ui5Yaml: 'ui5.yaml',
     Ui5LocalYaml: 'ui5-local.yaml',
     Ui5MockYaml: 'ui5-mock.yaml',
-    UI5DeployYaml: 'ui5-deploy.yaml'
+    UI5DeployYaml: 'ui5-deploy.yaml',
+    MtaYaml: 'mta.yaml'
```
Comment (Klaus-Keller) [inline packages/project-access/src/constants.ts:12]: Not want to be too picky, but the names are (mostly) in alphabetic order, would you mind move it below `ManifestAppDescrVar`.

File: packages/project-access/src/index.ts [hunk 2]
```diff
@@ -14,7 +14,8 @@ export {
     isCapNodeJsProject,
     loadModuleFromProject,
     readCapServiceMetadataEdmx,
-    readUi5Yaml
+    readUi5Yaml,
+    getMtaPath
```
Comment (Klaus-Keller) [inline packages/project-access/src/index.ts:18]: `getMtaPath` is also not inserted in alphabetic order :(.

File: packages/project-access/src/project/mta.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * @param projectPath Fiori app root folder
+ * @param fs
+ * @returns MtaPath
+ */
+export async function getMtaPath(projectPath: string, fs?: Editor): Promise<MtaPath | undefined> {
+    const mtaPath = await findFileUp(FileName.MtaYaml, projectPath, fs);
+    if (!mtaPath) {
+        return undefined;
+    } else {
+        console.log('before', mtaPath);
```
Comment (Klaus-Keller) [inline packages/project-access/src/project/mta.ts:23]: Is this `console.log` a leftover or should this remain in the code
```suggestion
```

File: packages/project-access/src/project/mta.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        const mtaFolderPath = dirname(mtaPath);
+        console.log('after', mtaFolderPath);
```
Comment (Klaus-Keller) [inline packages/project-access/src/project/mta.ts:25]: Same here, leftover or intention?
```suggestion
```

File: packages/project-access/src/project/mta.ts (hunk 3 above)
Comment (slin-sap) [reply packages/project-access/src/project/mta.ts:23]: > Is this `console.log` a leftover or should this remain in the code

Its a left over from debugging the unit test! I will remove it.

File: packages/project-access/src/constants.ts (hunk 1 above)
Comment (slin-sap) [reply packages/project-access/src/constants.ts]: Updated in alphabetic order

File: packages/project-access/src/index.ts (hunk 2 above)
Comment (slin-sap) [reply packages/project-access/src/index.ts]: Updated in alphabetic order

File: packages/project-access/src/project/mta.ts (hunk 4 above)
Comment (slin-sap) [reply packages/project-access/src/project/mta.ts]: Removed

**Review by Klaus-Keller — APPROVED** [review]
- Code is easy to understand and follows the guidelines
- Test coverage is great
- Did not build and test locally

Approved from my side.

File: packages/project-access/src/types/mta/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export interface MtaPath {
+    // File path that contains mta.yaml file
+    mtaPath: string;
+    // `hasRoot` is true if mta.yaml is in a parent folder (managed MTA project).
+    // It is set to false if mta.yaml is at in the project root folder (standardalone MTA project).
```
Comment (longieirl) [inline packages/project-access/src/types/mta/index.ts:5]: Different folder structures that can support mta.yaml;

managed approuter (1-1)
```
- fioiruiapp
- fioiruiapp -> mta.yaml
- fioiruiapp -> package.json
```

managed approuter (1-*)
```
- rootfolder
- rootfolder -> mta.yaml
- rootfolder -> package.json
- rootfolder -> fioriouiapp1 -> package.json
- rootfolder -> fioriouiapp2 -> package.json
```

standalone approuter
```
- rootfolder
- rootfolder -> mta.yaml
- rootfolder -> package.json
- rootfolder -> approuter -> package.json
- rootfolder -> fioriouiapp1 -> package.json
``

Comment (slin-sap) [reply packages/project-access/src/types/mta/index.ts:5]: @longieirl I think all scenarios are covered. Do you have any suggestions on the return type structure?

Comment (longieirl) [reply packages/project-access/src/types/mta/index.ts:5]: Managed approuter (1-*) is the same structure as standalone, so the only difference will be in the mta configuration where standalone will have a `approuter.nodejs`.

**Review by longieirl — APPROVED** [review]

**Pull request merged**